lib/internal/process/warning.js

```diff
@@ -1,9 +1,74 @@
 'use strict';
 
+const config = process.binding('config');
 const prefix = `(${process.release.name}:${process.pid}) `;
 
 exports.setup = setupProcessWarnings;
 
+var fs;
+var fd;
+var acquiringFd = false;
+function nop() {}
+
+function lazyFs() {
+  if (!fs)
+    fs = require('fs');
+  return fs;
+}
+
+function writeOut(message) {
+  if (console && typeof console.error === 'function')
+    return console.error(message);
+  process._rawDebug(message);
+}
+
+function onClose(fd) {
+  return function() {
+    lazyFs().close(fd, nop);
+  };
+}
+
+function onOpen(cb) {
+  return function(err, fd) {
+    acquiringFd = false;
+    if (fd !== undefined)
+      process.on('exit', onClose(fd));
+    cb(err, fd);
+    process.emit('_node_warning_fd_acquired', err, fd);
+  };
+}
+
+function onAcquired(message) {
+  return function(err, fd) {
+    if (err)
+      return writeOut(message);
+    lazyFs().appendFile(fd, `${message}\n`, nop);
```

Comment:
Since we ignore the errors, should we at-least document that this is just a best-effort functionality?